XTS/recovery/src/org/jboss/jbossts/xts/recovery/coordinator/CoordinatorRecoveryInitialisation.java

```diff
@@ -62,16 +63,20 @@ public static void startup()
             }
             
             try {
-                XTSRecoveryModule module = (XTSRecoveryModule)clazz.newInstance();
+                XTSRecoveryModule module = (XTSRecoveryModule)clazz.getDeclaredConstructor().newInstance();
                 module.install();
                 RecoveryManager.manager().addModule(module);
                 recoveryModules.add(module);
             } catch (InstantiationException ie) {
                 RecoveryLogger.i18NLogger.error_recovery_coordinator_CoordinatorRecoveryInitialisation_3(className, ie);
             } catch (IllegalAccessException iae) {
                 RecoveryLogger.i18NLogger.error_recovery_coordinator_CoordinatorRecoveryInitialisation_4(className, iae);
-            }
-        }
+            } catch (InvocationTargetException e) {
+                RecoveryLogger.i18NLogger.error_recovery_coordinator_CoordinatorRecoveryInitialisation_4(className, e);
+	    } catch (NoSuchMethodException e) {
+                RecoveryLogger.i18NLogger.error_recovery_coordinator_CoordinatorRecoveryInitialisation_4(className, e);
+	    }
```

Comment:
`error_recovery_coordinator_CoordinatorRecoveryInitialisation_4` is used to report `Unable to access module class`...I don't think you can reuse it for these other exceptions